design-documents/catalog-domain/catalog-service.md

```diff
@@ -0,0 +1,204 @@
+          ]
+        }
+      ]
+    },
+    {
+      "document": "attributes",
+      "filters": [
+        {
+          "field": "attributeCode",
+          "values": [
+            "attributeCode1",
+            "attributeCode2",
+            "attributeCode3"
+          ]
+        }
+      ],
+      "fields": [
+        {"field": "attributeCode"},
+        {"field": "values"}
+      ]
+    },
+    {
+      "document": "options",
+      "fields": [
+        {"field":  "*"}
+      ]
+    }
+  ]
+}
+```
+### XML
+
+```xml
+<query name="products">
+    <filters>
+        <field name="id">
+            <value>1652b2fe-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652b588-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652b7c2-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652b8f8-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652bb00-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652bc40-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652be5c-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652bf88-f758-11e9-8f0b-362b9e155667</value>
+            <value>1652c186-f758-11e9-8f0b-362b9e155667</value>
+            </value>
+        </field>
+    </filters>
+    <fields>
+    
+    </fields>
+    <field name="name" />
+    <field name="sku" />
+    <document name="priceRange" >
+        <document name="minimumPrice">
+            <field name="finalPrice" />
+        </document>
+    </document>
+    <document name="attributes">
+        <filter name="attributeCode">
+            <filter>attributeCode1</filter>
+            <filter>attributeCode2</filter>
+            <filter>attributeCode2</filter>
+        </filter>
+    </document>
+</query>
+```
+### GraphQL
+
+```graphql
+{
+  products(id: {in: [
+    "1652b2fe-f758-11e9-8f0b-362b9e155667",
+    "1652b588-f758-11e9-8f0b-362b9e155667",
+    "1652b7c2-f758-11e9-8f0b-362b9e155667",
+    "1652b8f8-f758-11e9-8f0b-362b9e155667",
+    "1652bb00-f758-11e9-8f0b-362b9e155667",
+    "1652bc40-f758-11e9-8f0b-362b9e155667",
+    "1652be5c-f758-11e9-8f0b-362b9e155667",
+    "1652bf88-f758-11e9-8f0b-362b9e155667",
+    "1652c186-f758-11e9-8f0b-362b9e155667"
+  ]}) {
+    fields {
+      name
+      sku
+      priceRange {
+        minimumPrice {
+          finalPrice
+        }
+      }
+    }
+  }
+}
+```
+
+# Query and scope
+
+Catalog service should support execution scope.
+The primary purpose of the scope is defining the data context.
+This context may lead to data transformation before returning data to a client.
+A good example of the scope usage is values localization.
+
+```json
+{
+  "query": "products",
+  "filters": [
+    {"field":  "id", "values": ["1652b2fe-f758-11e9-8f0b-362b9e155667"]}
+  ],
+  "fields": [
+    {"field": "name"},
+    {"field": "sku"}
+  ],
+  "scope" : [
+    {"argument":  "currency", "value":  "USD"},
+    {"argument":  "locale", "value":  "en_US"},
+    {"argument":  "store", "value":  "default"},
```

Comment:
It makes sense to change "argument" to "name" for consistency (see https://github.com/akaplya/architecture/blob/catalog-domain/design-documents/catalog-domain/product-data-object.md)